log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationSource.java

```diff
@@ -79,6 +98,24 @@ public ConfigurationSource(final InputStream stream, final URL url) {
         this.location = url.toString();
         this.file = null;
         this.data = null;
+        this.lastModified = 0;
+    }
+
+    /**
+     * Constructs a new {@code ConfigurationSource} with the specified input stream that originated from the specified
+     * url.
+     *
+     * @param stream the input stream
+     * @param url the URL where the input stream originated
+     * @param lastModified when the source was last modified.
+     */
+    public ConfigurationSource(final InputStream stream, final URL url, long lastModified) {
```

Comment:
So I'll note here that for 3.x, we should certainly be redoing this API. It exposes way too much internal state.

Reply:
Good luck with that. All this information has to be passed from one configuration to another to allow reconfiguration to occur.

Reply:
I've considered in the past porting over or implementing a subset of the [`Resource`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/Resource.html) interface from Spring. Do you think such an API would be more useful than having to inspect the ConfigurationSource union-like class?